OpenCL 2.2 graphics fixes

pierremoreau opened the pull request:
* Fixes compilation when setting GL_IS_SUPPORTED:
    * due to CMakeLists.txt referencing non-existent files and missing an include folder;
    * due to gl/main.cpp referencing an undefined variable.
* Add OpenGL tests that are present but were not referenced by CMakeLists.txt;
* Expose the graphics interop test toggles via CMake, allowing the user to enable/disable them without modifying CMakeLists.txt;
* Add such a toggle for OpenGL ES tests.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/OpenCL-CTS?pullRequest=3) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/OpenCL-CTS?pullRequest=3) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/OpenCL-CTS?pullRequest=3) it.</sub>

pierremoreau commented:
Is there anyone that could please review those changes?

mirh commented:
Ping. 
Is also anybody going to [fix 1.2](https://lists.freedesktop.org/archives/mesa-dev/2017-May/156971.html)?

pierremoreau commented:
@mirh I did had a try at fixing 1.2, starting from Aaron's patches and patches from other people, but given that it takes time, and it does not look like anyone is going to integrate those changes (or any changes) anyway, I have given up.

mirh commented:
@KhronosGroup ?

kpet commented:
@pierremoreau  @mirh  Sorry for the stupidly long delay. We're in the process of moving further development to Github to better be able to interact with the community. I should be able to get this in soon.

pierremoreau commented:
Thank you for the reply @kpet, and glad that things are moving along.

I was happy to see the OpenCL 2.2 rev. 8 Spec updates from yesterday fixing the issues I had reported earlier this year. :+1:

pierremoreau commented:
Friendly ping

kpet commented:
We're now ready to start accepting contributions, sorry again for the long delay. It looks like there's a conflict. Could you please rebase this?

pierremoreau commented:
Rebased done, please take another look.
The conflict was due to the changes regarding `issubnormal`; as it has been fixed upstream, I dropped the patch from this merge request.

pierremoreau commented:
By the way, are you looking at having a single branch for all OpenCL versions and have different test profiles for each version, or should I open a similar merge requests for 1.2, 2.0 and 2.1 as they have the same issues?

kpet commented:
Thanks, will have a look when the build completes.

>By the way, are you looking at having a single branch for all OpenCL versions and have different test profiles for each version

Yes, that's definitely the plan but it's going to take a while :). Before we start doing this, I'd like all vendors to upstream the bulk of the patches they still have on private branches.

>should I open a similar merge requests for 1.2, 2.0 and 2.1 as they have the same issues?

Yes, please.

pierremoreau commented:
> Yes, that's definitely the plan but it's going to take a while :). Before we start doing this, I'd like all vendors to upstream the bulk of the patches they still have on private branches.

That is great to hear! It will make it easier to keep things up to date, as well as running the CTS for 1.0 and 1.1, without needing the extra archives.

> Yes, please.

Okay, will do shortly.

kpet commented:
This now needs rebasing. Also, do you still intend to open PRs for the other branches?

pierremoreau commented:
I do, but have been lacking time to do so and when I tried last week, I ran into some compilation issue when compiling several of the trunk branches, partly due to not using the Git version of OpenCL-Headers, partly due to the cl12_trunk branch missing the `-Wno-c++11-narrowing` flag and clang 8.0 is now generating an error for that.

I rebased this branch and adding an extra fix due to additional changes in the code base since, but I haven’t been able to conform that it compiles due to a definition mismatch between glew.h and glext.h on my machine.

PS: Maybe we should add GL and DX interop compilation to Travis.

alycm commented:
I filed the `-Wno-c++11-narrowing` issue as #222.

kpet commented:
Thanks for the rebase. The changes look good. I'm going to try to cherry-pick it to other 2.x branches for you. I've created #223 so we don't lose track of the fact the extension tests aren't build tested.

pierremoreau commented:
Okay, I rebuilt glew from scratch so that it uses the latest OpenGL headers, and that got rid of the glew.h vs glext.h issues I had. I can confirm that the branch builds properly.

Thank you @alycm and @kpet for opening the extra issues.

I added two extra patches regarding enabling/disabling the graphics interop tests; the current CMake code was always unsetting them, overwriting whatever could be specified by the user when configuring the project. Also there was no such toggle for OpenGL ES tests.

I’ll work on the CL 2.0 and 2.1 MR now, and maybe the 1.2 if I solve some local config issues.

pierremoreau commented:
#224 opened for 2.1
#225 opened for 2.0
#226 opened for 1.2
